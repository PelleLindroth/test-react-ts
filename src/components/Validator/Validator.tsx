import { useState } from 'react'

const Validator = () => {
  const [name, setName] = useState<string>('')
  const [currentClass, setCurrentClass] = useState<string>('invalid')

  const validate = () => {
    if (name.match(/^[a-zA-Z\s]*$/)) {
      setCurrentClass('valid')
    } else {
      setCurrentClass('invalid')
    }

    setName('')
  }

  return (
    <form onSubmit={validate}>
      {name.length < 1 && <p>Please type your name</p>}
      <input
        className={currentClass}
        id="name"
        type="text"
        value={name}
        onChange={(e) => setName(e.target.value)}
      />
    </form>
  )
}

export default Validator
