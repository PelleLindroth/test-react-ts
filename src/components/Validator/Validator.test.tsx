import { render, screen } from '@testing-library/react'
import { shallow } from 'enzyme'
import userEvent from '@testing-library/user-event'
import Validator from './index'

describe('Validator', () => {
  it('renders component correctly', () => {
    const wrapper = shallow(<Validator />)

    expect(wrapper).toMatchSnapshot()
  })
  it('renders with empty text input field', () => {
    render(<Validator />)

    const textInput = screen.getByRole('textbox')

    expect(textInput).toHaveTextContent('')
  })
  it('sets input text class to invalid by default', () => {
    render(<Validator />)

    const textInput = screen.getByRole('textbox')

    expect(textInput).toHaveClass('invalid')
  })
  it('shows message while text input field is empty and hides it when typing', () => {
    render(<Validator />)

    const message = screen.getByText(/type/i)
    const textInput = screen.getByRole('textbox')

    expect(message).toBeInTheDocument()

    userEvent.type(textInput, 'iuey')

    expect(message).not.toBeInTheDocument()
  })
  it('changes input text class to valid when entering valid name', () => {
    render(<Validator />)

    const textInput = screen.getByRole('textbox')

    userEvent.type(textInput, 'Pelle Lindroth{enter}')

    expect(textInput).toHaveClass('valid')
    expect(textInput).not.toHaveClass('invalid')
  })
  it('changes input text class to invalid when entering invalid name', () => {
    render(<Validator />)

    const textInput = screen.getByRole('textbox')

    userEvent.type(textInput, 'Pelle Lindroth{enter}')

    expect(textInput).toHaveClass('valid')

    userEvent.type(textInput, 'P€££€{enter}')

    expect(textInput).toHaveClass('invalid')
  })
})
