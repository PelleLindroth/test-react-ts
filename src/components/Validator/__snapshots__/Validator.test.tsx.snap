// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator renders component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Please type your name
  </p>
  <input
    className="invalid"
    id="name"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
